osu.Game/Database/EFToRealmMigrator.cs

```diff
@@ -261,6 +262,12 @@ private void migrateBeatmaps(OsuDbContext ef)
                             var ruleset = r.Find<RulesetInfo>(beatmap.RulesetInfo.ShortName);
                             var metadata = getBestMetadata(beatmap.Metadata, beatmapSet.Metadata);
 
+                            if (ruleset == null)
+                            {
+                                log($"Skipping {++missing} beatmaps with missing ruleset");
```

Comment:
nit: it reads like it's skipping N beatmaps at the time of log, not that it skipped a total of N beatmaps. I think this reads better:
```suggestion
                                log($"Skipped a total of {++missing} beatmaps with missing ruleset");
```
(or instead log the total of skipped beatmaps separately after finish)